designs/0003-parallel_autodiff.md

```diff
@@ -0,0 +1,705 @@
+- Feature Name: parallel_ad
+- Start Date: 2019-04-20
+- RFC PR: (leave this empty)
+- Stan Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The goal of this RFC is to expand the CPU based parallelism
+capabilities of Stan math. This RFC proposes to introduce a new
+parallel map with automatic sharding and a parallel reduce which
+combines automatic sharding with vectorization.
+
+
+
+# Motivation
+[motivation]: #motivation
+
+Currently `map_rect` allows map parallelism, but it brings a few
+restrictions:
+
+1. Very user unfriendly due to the requirement to pack and unpack data
+   and parameters passed into the function.
+2. The user must provide a sensible sharding block size. No work load
+   balancing is done during execution.
+3. Vectorization must be traded for parallelism in most applications.
+
+In short, we currently make it very hard for users to use parallelism
+and moreover the current facility cannot speed up problems where we
+have to give up vectorization in exchange. 
+
+This RFC intends to make parallelism available to a much broader user
+base in a user-friendly way (addressing point 1+2) and that
+parallelism will scale much better as vectorization is combined with
+parallelism in a coherent approach (point 3).
+
+So ideally we can get more users in need for parallelism to actually
+use it and I also expect that vectorized parallelism will greatly
+enhance the problem set where parallelism actually speeds up a Stan
+program. For example, `map_rect` right now is not able to speed up
+programs which can take advantage of vectorization, but still have to
+process medium to large data sets. A parallel reduce with automatic
```

Comment:
The second part of this statement is supporting, but not really part of the design. This is implementation, right? Meaning, this doesn't actually get into the language part.